lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -478,43 +508,17 @@ bool TempRValueOptPass::tryOptimizeCopyIntoTemp(CopyAddrInst *copyInst) {
       auto *cai = cast<CopyAddrInst>(user);
       if (cai != copyInst) {
         assert(cai->getSrc() == tempObj);
-        if (cai->isTakeOfSrc() && !copyInst->isTakeOfSrc())
+        if (cai->isTakeOfSrc())
```

Comment:
I would add a comment here saying:

"If we have a copy addr user that is not our original copy inst, since our copy_addr passes in a +1 value, if we have a take, convert it to be a non-take. If we do a take of the source of our original copy src, we will handle it by inserting a destroy_addr in the appropriate place after this loop."

Reply:
I explained it in the comment above the loop